pkg/codegen/schema/schema.go

```diff
@@ -187,6 +187,17 @@ func (t *ObjectType) String() string {
 
 func (*ObjectType) isType() {}
 
+type ResourceType struct {
```

Comment:
Probably want the resource token (and perhaps the resource itself).

Reply:
Ha, completely forgot to circle back around to this. Will do.